app/src/test/java/com/mstudio/superheroarch/presentation/CharacterDetailViewModelTest.kt

```diff
@@ -0,0 +1,83 @@
+package com.mstudio.superheroarch.presentation
+
+import com.mstudio.superheroarch.RickAndMortyRepositoryInstruments
+import com.mstudio.superheroarch.presentation.detail.CharacterDetailViewModel
+import com.mstudio.superheroarch.presentation.detail.CharacterDetailViewTranslator
+import com.mstudio.superheroarch.presentation.network.NetworkManagerImpl
+import com.mstudio.superheroarch.remotedatasource.model.toCharacterData
+import com.mstudio.superheroarch.usecase.GetCharacterAndEpisodeUseCase
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.`when`
+import org.mockito.kotlin.verify
+
+@ExperimentalCoroutinesApi
+class CharacterDetailViewModelTest {
+
+    private lateinit var viewModel: CharacterDetailViewModel
+    private lateinit var view: CharacterDetailViewTranslator
+    private lateinit var useCase: GetCharacterAndEpisodeUseCase
+    private val testDispatcher = UnconfinedTestDispatcher()
+    private lateinit var networkManager: NetworkManagerImpl
+
+    @Before
+    fun before() {
+        Dispatchers.setMain(testDispatcher)
+        networkManager = mock()
+        useCase = mock()
+        view = mock()
+        `when`(networkManager.hasInternetConnection()).thenReturn(true)
+
+        viewModel = CharacterDetailViewModel(view, useCase, testDispatcher, networkManager)
+    }
+
+    @Test
+    fun `given a character detail screen, when screen is visited, show all the character info`() = runTest {
+
+        `when`(
+            useCase.getCharacterAndEpisode(
+                RickAndMortyRepositoryInstruments.givenACharacterRemoteEntity().toCharacterData()
+            )
+        ).thenReturn(RickAndMortyRepositoryInstruments.givenCharacterCompleteDetailData())
```

Comment:
Ideally you would store this object in a local variable to avoid generating it twice.

Reply:
I probably didn't explain too clearly. I meant you are calling `RickAndMortyRepositoryInstruments.givenCharacterCompleteDetailData()` twice. Usually in the tests you define the values once and then test that same value is returned. This instrument call could have some variability and calling it twice could make the test fail randomly.